src/public/app/widgets/buttons/global_menu.js

```diff
@@ -114,7 +153,52 @@ export default class GlobalMenuWidget extends BasicWidget {
         this.$widget.find(".open-dev-tools-button").toggle(isElectron);
         this.$widget.find(".switch-to-mobile-version-button").toggle(!isElectron);
 
+
         this.$widget.on('click', '.dropdown-item',
             () => this.$widget.find("[data-toggle='dropdown']").dropdown('toggle'));
+
+        this.loadUpdateAvailable();
+    }
+
+    async loadUpdateAvailable() {
+        const newVersion = await this.fetchNewVersion();
+
+        if (!newVersion) {
+            return;
+        }
+
+        const versionChange = GlobalMenuWidget.getVersionChange(CURRENT_VERSION, newVersion);
+        const showNewVersion = await this.checkIfShowNewVersionAvailable(versionChange);
+
+        if (!showNewVersion) {
+            return;
+        }
+
+        this.$widget.find(".global-menu-button-update-available").append(
+            new UpdateAvailableWidget()
+                .withVersionChange(versionChange)
+                .render()
+        )
+    }
+
+    async checkIfShowNewVersionAvailable(versionChange) {
+        const data = await server.get("options");
```

Comment:
There's `services/options.js` which already contains cached options, no need to fetch them from server.

Reply:
Which `options` do you mean? `app/services/options` or `services/options`?

Reply:
`service/options.js`